GDJS/Runtime/runtimeobject.ts

```diff
@@ -1640,40 +1684,134 @@ namespace gdjs {
       objectsLists: ObjectsLists,
       ignoreTouchingEdges: boolean
     ): boolean {
-      let moved = false;
-      let xMove = 0;
-      let yMove = 0;
+      let moveXArray: Array<float> =
+        RuntimeObject.separateFromObjectsStatics.moveXArray;
+      let moveYArray: Array<float> =
+        RuntimeObject.separateFromObjectsStatics.moveYArray;
+      moveXArray.length = 0;
+      moveYArray.length = 0;
+
       const hitBoxes = this.getHitBoxes();
+      const aabb = this.getAABB();
+
       for (const name in objectsLists.items) {
         if (objectsLists.items.hasOwnProperty(name)) {
-          const objects = objectsLists.items[name];
-
-          //Check if their is a collision with each object
-          for (let i = 0, len = objects.length; i < len; ++i) {
-            if (objects[i].id != this.id) {
-              const otherHitBoxes = objects[i].getHitBoxes();
-              for (let k = 0, lenk = hitBoxes.length; k < lenk; ++k) {
-                for (let l = 0, lenl = otherHitBoxes.length; l < lenl; ++l) {
-                  const result = gdjs.Polygon.collisionTest(
-                    hitBoxes[k],
-                    otherHitBoxes[l],
-                    ignoreTouchingEdges
-                  );
-                  if (result.collision) {
-                    xMove += result.move_axis[0];
-                    yMove += result.move_axis[1];
-                    moved = true;
-                  }
+          const otherObjects = objectsLists.items[name];
+
+          // Check if their is a collision with each object
+          for (const otherObject of otherObjects) {
+            if (otherObject.id === this.id) {
+              continue;
+            }
+            const otherHitBoxes = otherObject.getHitBoxesAround(
+              aabb.min[0],
+              aabb.min[1],
+              aabb.max[0],
+              aabb.max[1]
+            );
```

Comment:
The only impact should be in `SeparateFromObjects` where an AABB is asked on 1 object.
If the object has less than 4 hit boxes we could avoid to calculate its AABB and just call the "non optimized" method.